components/core/src/ffi/ir_stream/decoding_methods.hpp

```diff
@@ -108,22 +109,23 @@ namespace ffi::ir_stream {
      */
     IRErrorCode get_encoding_type (IrBuffer& ir_buf, bool& is_four_bytes_encoding);
 
-    namespace eight_byte_encoding {
-        /**
-         * Decodes the preamble for the eight-byte encoding IR stream.
-         * @param ir_buf
-         * @param ts_info Returns the timestamp info on success
-         * @return IRErrorCode_Success on success
-         * @return IRErrorCode_Corrupted_IR if ir_buf contains invalid IR
-         * @return IRErrorCode_Incomplete_IR if ir_buf doesn't contain enough
-         * data to decode
-         * @return IRErrorCode_Unsupported_Version if the IR uses an unsupported
-         * version
-         * @return IRErrorCode_Corrupted_Metadata if the metadata cannot be
-         * decoded
-         */
-        IRErrorCode decode_preamble (IrBuffer& ir_buf, TimestampInfo& ts_info);
+    /**
+     * Decodes the preamble for the eight-byte encoding IR stream.
+     * @param ir_buf
+     * @param metadata_type Returns the type of the metadata found in the IR
+     * @param metadata_pos Returns the starting position of the metadata in ir_buf
+     * @param metadata_size Returns the size of the metadata written in the IR
+     * @return IRErrorCode_Success on success
+     * @return IRErrorCode_Corrupted_IR if ir_buf contains invalid IR
```

Comment:
This is only possible if `read_metadata_info` returns it. The only case `read_metadata_info` returns it is if the type is not JSON, which we probably shouldn't validate anymore since the FFI layer doesn't care about the contents of the metadata anymore.

Reply:
Whoops, missed the case where it returns corrupted when the length is wrong. But the JSON validation point still remains true.